### ISSUE TITLE
Error Message: Invalid Argument

### ISSUE THREAD

**Issue by orwant** [google/google-visualization-issues#489]
```
I have been using Alexander Bautz Google Visualisation suite for SharePoint for the
past few months successfully for our online dashboards.

Today some of the graphs failed to resolve on the page producing the Invalid Argument
error message.  This seem to have been caused when either column or bar charts are
trying to be produced with a large number of rows.

The data set has not changed and yesterday - 11/01/2010 - the graphs would resolve
correctly and today there is the error.

I have resolved the issue by rolling back the implementation to version 1 of the API
from version 1.1 in the initialisation script.

Can you confirm if a change had been made to the environment in the last 24 hours which
would cause such an issue?



*********************************************************
For developers viewing this issue: please click the 'star' icon to be
notified of future changes, and to let us know how many of you are
interested in seeing it resolved.
*********************************************************

```

Original issue reported on code.google.com by `jgpmckeogh` on 2011-01-11 10:31:10

**Comment by orwant** [comment]
```
Yes, we have launched a new release to the 1.1 version at that time, so this is probably
the cause. About 24 hours later we fixed some issues are refreshed the 1.1 version,
so the issue you described might be gone now. If not, can you please elaborate on the
problem, because I didn't really understand it. It would be best if you could send
a small demo page that demonstrates the p[problem.

Thanks.

```

Original issue reported on code.google.com by `jinji.viz` on 2011-01-12 12:51:35

**Comment by orwant** [comment]
```
Hi,

we are probably having the same problem, using 1.1 version. Our column charts started
to show Invalid Argument error, I have found some older issue describing what I think
is the same problem with IE here: http://code.google.com/p/google-visualization-api-issues/issues/detail?id=373
although the example page is working alright since google visualization gallery is
using 1.0 version.
I'm attaching demo file with 1 working and 1 not working column chart, both are beaing
created the same way and both should have the same amount of columns.


Hope this helps
Jan Ptacek
```

Original issue reported on code.google.com by `ptacek.j` on 2011-01-12 16:29:18

<hr>
- _Attachment: [test.html](https://storage.googleapis.com/google-code-attachments/google-visualization-api-issues/issue-489/comment-2/test.html)_

**Comment by orwant** [comment]
```
I don't understand. Is the problem in IE only or in all browsers?
Anyway, I tried the test.html and it worked identically for 1.0 and 1.1, both on Chrome
and on IE8.

Can you elaborate?

```

Original issue reported on code.google.com by `jinji.viz` on 2011-01-12 17:26:37

**Comment by orwant** [comment]
```
only in IE. 
I know i had this problem with IE7 and IE8, the first chart in test.html should show
invalid argument error.
```

Original issue reported on code.google.com by `ptacek.j` on 2011-01-12 17:33:51

**Comment by orwant** [comment]
```
I tried to find when exactly I'm getting this error and i've found that if there is
0 value in data, i get "invalid argument".
```

Original issue reported on code.google.com by `ptacek.j` on 2011-01-13 12:55:41

**Comment by orwant** [comment]
```
Can you provide a complete example? I can't reproduce it otherwise...
```

Original issue reported on code.google.com by `jinji.viz` on 2011-01-13 16:59:20

**Comment by orwant** [comment]
```
I'm having the same issue. I get it only in...

1) IE8
2) bar graphs
3) with values of either 0, or when the scaling is so great it is "like" 0 (i.e. a=1,000,000
b=10 throws an error).

-Jason
```

Original issue reported on code.google.com by `jason.hooten` on 2011-01-19 21:52:27

**Comment by orwant** [comment]
```
This is a serious bug in IE for column bar chart because it cannot display either 0
or small value when other values are larger. Mozilla firefox is fine so this is IE8
problem with Column chart. Simple example is shown below.

=================================================================
<html>   <head>     <script type="text/javascript" src="https://www.google.com/jsapi"></script>
    <script type="text/javascript">       google.load("visualization", "1", {packages:["corechart"]});
      google.setOnLoadCallback(drawChart);       function drawChart() {         

var data = new google.visualization.DataTable();         
data.addColumn('string', 'Year');         
data.addColumn('number', 'Sales');         

data.addRows(4);         
data.setValue(0, 0, '2004');         
data.setValue(0, 1, 2);                
data.setValue(1, 0, '2005');         
data.setValue(1, 1, 1);              
data.setValue(2, 0, '2006');         
data.setValue(2, 1, 0);              
data.setValue(3, 0, '2007');         
data.setValue(3, 1, 10);            

var chart = new google.visualization.ColumnChart(document.getElementById('chart_div'));

chart.draw(data, {width: 400, height: 240, title: 'Company Performance',          
                hAxis: {title: 'Year', titleTextStyle: {color: 'red'}}            
             });       }     </script>   </head>    <body>     

<div id="chart_div"></div>   </body> </html>

```

Original issue reported on code.google.com by `MarketFellow` on 2011-01-19 21:59:42

**Comment by orwant** [comment]
```
I'm having the same issue. The problem occurs when the "data.setValue" are of "0".

IE 7,8 -> "Invalid argument"

```

Original issue reported on code.google.com by `ing.michal.kozel` on 2011-01-19 22:10:00

<hr>
- _Attachment: [example.txt](https://storage.googleapis.com/google-code-attachments/google-visualization-api-issues/issue-489/comment-9/example.txt)_

**Comment by orwant** [comment]
```
I feel much better knowing I'm not the only one.

-Jason
```

Original issue reported on code.google.com by `jason.hooten` on 2011-01-19 22:17:06

**Comment by orwant** [comment]
```
google.visualization.Query.setResponse({'version':'0.5', 'reqId':'', 'status':'OK',
'table': {"cols": [{"id":"string","label":"Name","type":"string"},{"id":"number","label":"Burts","type":"number"}],"rows":
[{ "c":[{"v":"Inc"},{"v":0}]},{ "c":[{"v":"Res"},{"v":1}]}]}}); 

the response above has the error while

google.visualization.Query.setResponse({'version':'0.5', 'reqId':'', 'status':'OK',
'table': {"cols": [{"id":"string","label":"Name","type":"string"},{"id":"number","label":"Burts","type":"number"}],"rows":
[{ "c":[{"v":"Inc"},{"v":3}]},{ "c":[{"v":"Res"},{"v":3}]}]}}); 

does not.
IE8 only. firefox is fine.
```

Original issue reported on code.google.com by `bquirion` on 2011-01-19 22:32:44

**Comment by orwant** [comment]
```
Whose mistake is this? Google or IE? 

I too am facing the same issue and I wonder what answer will I give to my client.

Subash.
```

Original issue reported on code.google.com by `n.subashkumar` on 2011-01-20 08:56:34

**Comment by orwant** [comment]
```

We have identified the problem and working on fixing this issue as soon as possible.
```

Original issue reported on code.google.com by `metula` on 2011-01-20 09:09:30

**Comment by orwant** [comment]
```
Issue 497 has been merged into this issue.
```

Original issue reported on code.google.com by `metula` on 2011-01-20 09:13:49

**Comment by orwant** [comment]
```
Issue 498 has been merged into this issue.
```

Original issue reported on code.google.com by `metula` on 2011-01-20 09:15:20

**Comment by orwant** [comment]
```
We are using gwt-visulaization(core chart) api for rendering charts in our real time
project. The column chart was working fine for us till yesterday and when we are testing
it today we are getting invalid argument error for column chart. However pie chart
is rendering absoultely fine.

We are using gwt-visulaization 1.1 version for this implementation.We are in the midst
of UAT and the project is expected to go live by next week and now we are in a critical
situation and without this the project might not go live.

Any help from your side to fix this issue immediately would be of immense help. 
```

Original issue reported on code.google.com by `veeraraghavan.kannan` on 2011-01-20 11:46:51

**Comment by orwant** [comment]
```
Can you please roll back the production version to a few days ago when it was working
until the issue can be fixed and proven out?  Many people rely on these charts and
right now we are left with nothing. If you don't roll it back, we are all stuck with
no charts until it is fixed.  There should be no issue putting it back where it was
right?

thanks for your help,
-Matt
```

Original issue reported on code.google.com by `MattMaltarich` on 2011-01-20 13:31:41

**Comment by orwant** [comment]
```
You can always go back to using version 1.0, it works fine, and some new features are
already in it (chartArea i.e.), I was forced to do that myself and I had no problem
since.
```

Original issue reported on code.google.com by `ptacek.j` on 2011-01-20 13:46:10

**Comment by orwant** [comment]
```
I have always been on version 1.  All of my charts went blank as of yesterday when they
contain a "0" value.  Switching between 1 and 1.1 does not seem to change anything,
and I have never been on 1.1

thanks,
-Matt
```

Original issue reported on code.google.com by `MattMaltarich` on 2011-01-20 13:53:15

**Comment by orwant** [comment]
```
I have the same problems with zero values in IE. I changed 
google.load('visualization','1',{packages: ['corechart']}); 
to 
google.load('visualization','1.0',{packages: ['corechart']});
and
google.load('visualization','1.1',{packages: ['corechart']});
but the problem remains in IE.

Thanks.
```

Original issue reported on code.google.com by `arley.ferreira.souza` on 2011-01-20 14:45:02

**Comment by orwant** [comment]
```
I'm sorry, I thought that changing to 1.0 should solve the problem. 
When I found this error I was using 1.1 and when I switched back to 1.0 it was OK,
now it looks like 1.0 was updated with 1.1 and this BUG was deployed with it.

Now we'll have to wait.
```

Original issue reported on code.google.com by `ptacek.j` on 2011-01-20 14:53:19

**Comment by orwant** [comment]
```
Would appreciate a fix to this also ASAP. 

Perhaps a feature request for versioning on the API so that we aren't subject to these
sorts of problems (and others) introduced each time Google release a new version of
the API. 
```

Original issue reported on code.google.com by `tom.betts` on 2011-01-20 16:25:52

**Comment by orwant** [comment]
```
Please fix ASAP, and test with IE next time!!!
```

Original issue reported on code.google.com by `denicomp` on 2011-01-20 16:45:12

**Comment by orwant** [comment]
```
I'm also having this problem.  Thanks for your guys' hard work to get things like this
fixed quickly, that's what I love about using Google tools, we can rest assured this
will be fixed soon!
```

Original issue reported on code.google.com by `dmp013` on 2011-01-20 18:09:45

**Comment by orwant** [comment]
```
For the record, we are using 1.0 (never moved to 1.1) and are also experiencing the
same "Invalid Argument" error in IE. OK in FF and Chrome.

Anxiously awaiting the fix.
```

Original issue reported on code.google.com by `Roger1of6` on 2011-01-20 18:13:33

**Comment by orwant** [comment]
```
The same problem, we didn´t change to 1.1 and we have the same problem. Ok in FF, but
not in explorer.
```

Original issue reported on code.google.com by `luis.atilag` on 2011-01-20 18:19:07

**Comment by orwant** [comment]
```
I switched to 1.1 and this issue resolved itself.
```

Original issue reported on code.google.com by `dmp013` on 2011-01-20 18:26:14

**Comment by orwant** [comment]
```
I lied, switching to 1.1 fixed the issue in SOME cases, but I'm still seeing it pop
up on other pages.
```

Original issue reported on code.google.com by `dmp013` on 2011-01-20 18:42:36

**Comment by orwant** [comment]
```
It seems the 0 issue is fixed but not the like Zero. Here is a sample Json object for
the like zero case.

{"cols": [
   {
     "id":"string","label":"Name","type":"string"},
     "id":"number","label":"Burts","type":"number"
   }],
"rows": [
   { "c":[{"v":"Inc"},{"v":10}]},
   { "c":[{"v":"Res"},{"v":1000000}]}
  ]
}; 

```

Original issue reported on code.google.com by `jason.hooten` on 2011-01-20 18:42:48

**Comment by orwant** [comment]
```
The problem is fixed in IE8 as I verified. But I worry more about quality of Google
library for this kind of error. Is there a regression setup to run on IE for code changes?
When there are many users to use this public API, it is very important to secure quality
and stability of this tool. Thanks for help to resolve this issue.Best wishes for google
API.
```

Original issue reported on code.google.com by `ning.jiang.usa` on 2011-01-20 19:09:02

**Comment by orwant** [comment]
```
Appears to be fixed for me now also.  Fortunately (or unfortunately) I found the problem
20 minutes before a demo... 
```

Original issue reported on code.google.com by `denicomp` on 2011-01-20 20:36:42

**Comment by orwant** [comment]
```
It's seems to work on some of my charts but not others.  I have not identified the difference
yet.  All of my charts work as they did before the change if I load them in Firefox
or Google Chrome.  IE still has some issues.  

Is there a problem with rolling back to the previous version 1.0 so that we have a
stable version until this is completely solved?  That should be easy right?

We can all test V1.1 once the fix is done and if it fully works, then move it to 1.0

-Matt
```

Original issue reported on code.google.com by `MattMaltarich` on 2011-01-20 20:40:28

**Comment by orwant** [comment]
```
The problem should be completely resolved by now. Please let us know if there are any
issues still.
```

Original issue reported on code.google.com by `metula` on 2011-01-20 23:35:40

**Comment by orwant** [comment]
```
We are validating the charts. Tomorrow once we deploy this code into our production
system and if there is a change pushed in by your team which causes some problem and
if charts are not rendered then we will be fired and we cannot go to the customer telling
there is an issue with the jar which the gwt-visulazation team is working on.

My question is why is this connecting to google at first place, we are using as a jar
and it should be standalone one am I right? secondly, is there any other chart tool
which gwt supports and is not tightly coupled with google? Please advise.

thanks,
-veera

```

Original issue reported on code.google.com by `veeraraghavan.kannan` on 2011-01-21 04:03:59

**Comment by orwant** [comment]
```
The problem isn't solved !

On my bar charts, I have always error messages :

- "'undefined' a la valeur Null ou n'est pas un objet." with IE 

- "Cannot read property 'type' of undefined" with Chrome.

Works fine with Firefox.
```

Original issue reported on code.google.com by `thomas.cousin56` on 2011-01-21 08:11:10

**Comment by orwant** [comment]
```

Thomas, the issue you describe here seems unrelated to this topic. Please open a separate
issue and provide some more details/example that causes this.

Veera, the gwt jar that you are loading is using the online visualization code that
is downloaded every time you are rendering a chart on a page. We serve at any given
point in time the the normal version (1.0) and the release candidate (1.1).
```

Original issue reported on code.google.com by `metula` on 2011-01-21 08:47:55

**Comment by orwant** [comment]
```
I believed that it was the same problem because my bar charts don't work since yesterday
on IE and Chrome.

It seems that someone else have the same issue as me (see issue 501).
```

Original issue reported on code.google.com by `thomas.cousin56` on 2011-01-21 08:54:10

**Comment by orwant** [comment]
```
I'm still having this problem with column charts in IE8.  Have tried ver 1 and 1.1 to
no avail.  Here is anm example of a chart which throws "Invalid argument" error:

google.setOnLoadCallback(drawChart);
function drawChart() {
  var data = google.visualization.arrayToDataTable([
    ["Age","Clients","Not Offered","Accepted Offer","Declined Offer"],
    ["16",24,7,18,4],
    ["17",26,8,14,8],
    ["18",6,2,2,2],
    ["19",2,0,2,0],
    ["20",2,2,0,0],
    ["21",2,1,1,0],
    ["22",1,1,0,0]
  ]);
  new google.visualization.ComboChart(document.getElementById('gc-referrals')).draw(data,
{
    width:643, 
    height:400, 
    chartArea:{width:543, height:300}, 
    legend:'top',
    seriesType: 'bars',
    isStacked:true,
    series: {0: {type: 'line'}}
  });
}
```

Original issue reported on code.google.com by `localcause.pg` on 2011-06-30 11:53:08

**Comment by orwant** [comment]
```
Sorry, technically this is a combo chart not a column chart, but I assume it draws it's
columns using the same code etc.
```

Original issue reported on code.google.com by `localcause.pg` on 2011-06-30 11:56:33

**Comment by orwant** [comment]
```
Ah, technicalities... Unlike ColumnChart, ComboChart doesn't support isStacked at the
moment. See http://code.google.com/apis/chart/interactive/docs/gallery/combochart.html#Configuration_Options.
Also, please don't add comments to an issue long closed. You can open a new one. Chances
are this is really a different issue.
```

Original issue reported on code.google.com by `jinji.viz` on 2011-06-30 14:04:11

**Comment by orwant** [comment]
```
Pie Chart Still Not Working in any Of ie version in standard mode...attached html file
is from google resource only plz help..
```

Original issue reported on code.google.com by `puravmehta12` on 2013-08-21 05:36:41

<hr>
- _Attachment: [chart.html](https://storage.googleapis.com/google-code-attachments/google-visualization-api-issues/issue-489/comment-41/chart.html)_
